File: solutions/go/0049-group-anagrams.go
package main

import (
	"sort"
	"strings"
)

func sortString(s string) string {
	str := strings.Split(s, "")
	sort.Strings(str)

	return strings.Join(str, "")
}

func groupAnagrams(strs []string) [][]string {
	anagramGroups := make(map[string][]string, 0)
	for _, word := range strs {
		key := sortString(word)
		anagramGroups[key] = append(anagramGroups[key], word)
	}

	groups := make([][]string, 0, len(anagramGroups))
	for _, group := range anagramGroups {
		groups = append(groups, group)
	}

	return groups
}

File: solutions/php/0049-group-anagrams.php
<?php
class Solution
{
    /**
     * @param String[] $strs
     * @return String[][]
     */
    function groupAnagrams(array $strs): array
    {
        $anagrams = [];

        foreach ($strs as $i => $word) {
            $anagrams[$this->sortStr($word)][] = $word;
        }

        return $anagrams;
    }

    private function sortStr(string $str): string
    {
        $arr = str_split($str);
        sort($arr);

        return implode($arr);
    }
}
